ecosystem/nft-metadata-crawler/src/asset_uploader/api/upload_batch.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    asset_uploader::api::BatchUploadRequest,
+    models::asset_uploader_request_statuses::AssetUploaderRequestStatuses, schema,
+};
+use ahash::AHashMap;
+use anyhow::Context;
+use diesel::{
+    r2d2::{ConnectionManager, Pool, PooledConnection},
+    BoolExpressionMethods, ExpressionMethods, PgConnection, QueryDsl, RunQueryDsl,
+};
+use tracing::debug;
+use url::Url;
+use uuid::Uuid;
+
+/// Uploads a batch of assets to the asset uploader worker
+pub fn upload_batch(
+    pool: Pool<ConnectionManager<PgConnection>>,
+    request: &BatchUploadRequest,
+) -> anyhow::Result<String> {
+    let mut conn = pool.get()?;
+    let existing_rows = get_existing_rows(&mut conn, &request.urls)?;
+
+    let request_id = Uuid::new_v4();
+    let application_id = Uuid::parse_str(&request.application_id)?;
```

Comment:
i think this service doesn't sit behind ag? otherwise, we might need to use internal routing (to be built still) for nft crawler as well

Reply:
the tmp backend should provide the app id